src/sparse_approximations.jl

```diff
@@ -8,20 +22,39 @@ struct ApproxPosteriorGP{Tapprox,Tprior,Tdata} <: AbstractGP
 end
 
 """
-    approx_posterior(::VFE, fx::FiniteGP, y::AbstractVector{<:Real}, u::FiniteGP)
+    posterior(v::VFE, fx::FiniteGP, y::AbstractVector{<:Real})
+
+Compute the optimal approximate posterior [1] over the process `f = fx.f`, given observations `y`
+of `f` at `x`, and inducing points `v.z`.
+
+```jldoctest
+julia> f = GP(Matern52Kernel());
+
+julia> x = randn(1000);
+
+julia> z = range(-5.0, 5.0; length=13);
+
+julia> v = VFE(f(z));
+
+julia> y = rand(f(x, 0.1));
 
-Compute the optimal approximate posterior [1] over the process `f`, given observations `y`
-of `f` at `x`, and inducing points `u`, where `u = f(z)` for some inducing inputs `z`.
+julia> post = posterior(v, f(x, 0.1), y);
+
+julia> post(z) isa AbstractGPs.FiniteGP
+true
+```
 
 [1] - M. K. Titsias. "Variational learning of inducing variables in sparse Gaussian
 processes". In: Proceedings of the Twelfth International Conference on Artificial
 Intelligence and Statistics. 2009.
 """
-function approx_posterior(::VFE, fx::FiniteGP, y::AbstractVector{<:Real}, u::FiniteGP)
+function posterior(vfe::VFE, fx::FiniteGP, y::AbstractVector{<:Real})
+    @assert vfe.fz.f == fx.f
```

Comment:
```suggestion
    @assert vfe.fz.f === fx.f
```
Correct me if I'm wrong on this, but I suspect that we want to ensure that it's _literally_ the same object.